Getting Started with Directus and Flutter (#140)
directus-labs/guest-authoring: crud-with-flutter-and-directus/index.md

                } else if (pagesSnapshot.hasError) {
                  return Text('Error: ${pagesSnapshot.error}');
                } else {
                  final pages = pagesSnapshot.data!;
                  return pages.isNotEmpty
                      ? PageWidget(
                          page: pages,
                        )
                      : const Text('No pages found');
                }
              },
            ),
          );
        }
      },
    );
  }
}
```

![Showing the contents from the pages collection in flutter application](./Screenshot%202024-04-17%20at%2017.41.14.png)

  
## Creating Blog Posts With Directus

Similar to creating pages, you can also create and manage blog posts using Directus CMS. To do that, create a new collection named posts with the following fields:
  
- **slug**: Primary key field, Manually entered string
- **title**: Text input field
- **content**: WYSIWYG input field
- **status**: Dropdown filed
- **date_created**: Datetime field
    
![Creating the posts collection in Directus CMS](./Screenshot%202024-04-17%20at%2011.13.45.png)

Now add a couple of blog data to the `posts` collection.
    

> ![Adding entries to Directus collection](./Screenshot%202024-04-17%20at%2011.26.32.png)

Author: Done

  });

  @override
  Widget build(BuildContext context) {
    return GestureDetector(
      onTap: () {
        Navigator.push(
          context,
          MaterialPageRoute(
            builder: (context) => BlogPostWidget(post: blogPost),
          ),
        );
      },
      child: Container(
        margin: const EdgeInsets.only(bottom: 16.0),
        child: Column(
          crossAxisAlignment: CrossAxisAlignment.start,
          children: [
            Text(
              blogPost['title'],
              style: Theme.of(context).textTheme.labelLarge,
            ),
          ],
        ),
      ),
    );
  }
}
```
With the above code, when the user taps on the `BlogPostItem`, it triggers the `onTap` callback function. Inside this function, we use `Navigator.push` to navigate to a new screen. We create a `MaterialPageRoute`, defining the widget to be displayed on the new screen as `BlogPostWidget`. Also, we pass the `blogPost` data as a parameter to the `BlogPostWidget` widget. This allows us to display detailed information about the selected `blog` post on the new screen.
    
![Navigating to the blog single page](./Screenshot%202024-04-17%20at%2018.27.22.png)

## Summary

> Throughout this tutorial, you've learned how to build a CRUD application using Directus and Flutter. We started by creating a new project, set up environment variables and everything we need to call Directus. Then we went further to create pages, and posts collections in Directus and integrate the Flutter application with Directus by making API requests to perform CRUD operations.
> 
> Now that you've acquired this knowledge, what would you like to build next with the Directus CMS? Perhaps you can improve on this project by adding authentication to protect the application, and form validation/serialization. To learn more about Dirctus and other interesting features it provides, visit the [official documentation](https://docs.directus.io/).

Reviewer: ```suggestion
Throughout this tutorial, you've learned how to build a Flutter application that uses data from a Directus project. We started by creating a new project, set up environment variables and everything we need to call Directus. We then created pages and posts collections in Directus and integrated them with the the Flutter.
```
